tests/test_filterDiaSourceCatalog.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of ap_association.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import lsst.utils.tests
+import lsst.meas.base.tests as measTests
+import lsst.geom as geom
+from lsst.ap.association.filterDiaSourceCatalog import (FilterDiaSourceCatalogConfig,
+                                                        FilterDiaSourceCatalogTask)
+
+
+class TestFilterDiaSourceCatalogTask(unittest.TestCase):
+
+    def setUp(self):
+        self.nSources = 10
+        self.yLoc = 100
+        self.expId = 4321
+        self.bbox = geom.Box2I(geom.Point2I(0, 0),
+                               geom.Extent2I(1024, 1153))
+        dataset = measTests.TestDataset(self.bbox)
+        for srcIdx in range(self.nSources):
+            dataset.addSource(10000.0, geom.Point2D(srcIdx, self.yLoc))
+        schema = dataset.makeMinimalSchema()
+        schema.addField("sky_source", type="Flag", doc="Sky objects.")
+        _, self.diaSourceCat = dataset.realize(10.0, schema, randomSeed=1234)
+        self.diaSourceCat[0:5]["sky_source"] = True
+        self.config = FilterDiaSourceCatalogConfig()
+
+    def test_run(self):
+        self.config.doRemoveSkySources = False
+        filterDiaSourceCatalogTask = FilterDiaSourceCatalogTask(config=self.config)
+        result = filterDiaSourceCatalogTask.run(self.diaSourceCat, ccdVisitId=self.expId)
+        self.assertEqual(len(result.filteredDiaSourceCat), len(self.diaSourceCat))
+
+    def test_run_no_filter(self):
+        self.config.doRemoveSkySources = True
+        filterDiaSourceCatalogTask = FilterDiaSourceCatalogTask(config=self.config)
+        result = filterDiaSourceCatalogTask.run(self.diaSourceCat, ccdVisitId=self.expId)
+        self.assertEqual(len(result.filteredDiaSourceCat),
```

Comment:
Add an additional couple `assertEqual`s comparing the total number of returned sources to `self.nSources - self.nSkySources`, and the number of sources in the returned (that I'd like you to add) sky sources catalog to `self.nSkySources`

Reply:
I've added an assertEqual for your first suggestion. I'm a bit unclear on the second suggestion- mostly confused by the `that I'd like you to add`

Reply:
See my above comment about adding an optional output to your task that returns the catalog of sources that have been removed.